feat: Add base CoreWebVitals view

rafaeelaudibert opened the pull request:
This PR started with the top navbar for Core Web Vitals, but because our squash workflow kinda sucks with stacked PRs, I built it all in a single PR. Here's what it looks like:

https://www.loom.com/share/e5733b1758564feb9cd89f29f7b33b57?sid=feeb7af4-0ff1-4f47-bca0-3836d6728a5dhttps://www.loom.com/share/e5733b1758564feb9cd89f29f7b33b57?sid=feeb7af4-0ff1-4f47-bca0-3836d6728a5d

Supersedes #27449

rafaeelaudibert committed:
feat: Add basic CoreWebVitals node

rafaeelaudibert committed:
style: Add basic CoreWebVitals view

Next commit will iron out this UI to make it look nicer, but the basic of the functionality is here already.

rafaeelaudibert committed:
style: Style the top layer for core web vitals

This is what the top layer will look like once we have our four main metrics. In the future we might decide to expand for more metrics, and in that case a side panel will make more sense, but this is enough for a first launch - hopefully next week.

rafaeelaudibert committed:
fix: Small changes

rafaeelaudibert committed:
fix: Fix schema

Had to update my local schema gen library to get it to work, sorry

rafaeelaudibert commented on posthog/schema.py:
```diff
@@ -428,6 +428,19 @@ class ConditionalFormattingRule(BaseModel):
     templateId: str
 
 
+class Math(StrEnum):
```
This is a terrible name. It's called `CoreWebVitalsPercentile` in my code, so I'm not sure why is it that it's generating this terribly-named `Math` class

rafaeelaudibert committed:
fix: Remove class after change on Tooltip

rafaeelaudibert committed:
feat: Add Core Web Vitals health check

rafaeelaudibert committed:
feat: Move core web vitals tab selection to logic

Remove inner `useState`, we'll need to use that state in the logic

rafaeelaudibert commented on frontend/__snapshots__/scenes-app-insights--funnel-top-to-bottom-breakdown-edit--light.png:
I don't believe any of my changes could've caused this layout to shift 🤔

rafaeelaudibert replied:
Snapshot was updated for the third time now, god

rafaeelaudibert committed:
feat: Add `annotations` feature to our `LineGraph`s

Inside `insightViz` we can now specify custom annotation lines for our charts

rafaeelaudibert committed:
feat: Add annotations to Core Web Vitals chart

rafaeelaudibert committed:
feat: Display `duration_ms` with one decimal house

rafaeelaudibert committed:
feat: Add CoreWebVitalsContent

This is the inner part of the Core Web Vitals view, the UI is now mostly complete, we're only missing the split by paths to get UI parity with Vercel (we'll miss some of the 0-100 analysis but that's harder to do, we'll release that eventually)

rafaeelaudibert committed:
refactor: Split CoreWebVitals component in separate files

rafaeelaudibert commented:
I'll have to investigate why is it that we have an infinite loop when generating our `webAnalyticsLogicType.ts` file. This happens locally as well. I need to extract that to a separate `coreWebVitalsLogic.ts` file anyway, so I might go that route.

rafaeelaudibert committed:
fix: Add type to query to fix typegen

rafaeelaudibert committed:
fix: Check for tick value before rendering

robbie-c commented on frontend/src/queries/schema/schema-general.ts:
```diff
@@ -659,9 +664,10 @@ export interface DataTableNode
 }
 
-export interface GoalLine {
+export interface ChartAnnotation {
```
There's a different usage of the word Annotation in charts, we use it in trends for github PR merges. (this thing)
![Uploading Screenshot 2025-01-15 at 18.49.59.png…]()

Please use a different name :D or keep to GoalLine

rafaeelaudibert replied:
Ah lol, sure. It sucks that `chart.js` calls it annotations though. Will revert to `GoalLine` :)

robbie-c commented on frontend/src/queries/utils.ts:
```diff
@@ -341,6 +347,13 @@ export const getResultCustomizationBy = (query: InsightQueryNode): ResultCustomi
     return undefined
 }
 
+export const getAnnotations = (query: InsightQueryNode): ChartAnnotation[] | undefined => {
```
similar about the word annotation

robbie-c reviewed (approved):
All good if you change the name :)

rafaeelaudibert committed:
refactor: Rename `annotations` to `goalLine`

We already have the concept of `annotations` on charts, so let's use `goalLines` rather than `annotations` to make the code slightly clearer

rafaeelaudibert committed:
fix: Add missing refactor

Refactored this in last PR but forgot to commit

rafaeelaudibert committed:
fix: Fix remaining `annotations` code

rafaeelaudibert committed:
fix: TS errors

github-actions[bot] committed:
Update query snapshots
